qt/aqt/browser/browser.py

```diff
@@ -862,10 +862,13 @@ def set_due_date(self) -> None:
     @skip_if_selection_is_empty
     @ensure_editor_saved
     def forget_cards(self) -> None:
-        forget_cards(
+        if op := forget_cards(
             parent=self,
             card_ids=self.selected_cards(),
-        ).run_in_background()
+            restore_position_key=Config.Bool.RESTORE_POSITION_BROWSER,
+            reset_counts_key=Config.Bool.RESET_COUNTS_BROWSER,
```

Comment:
This doesn't scale very well. With every new option, callers need to pass in one more argument.
Maybe there should be a `ConfigContext` enum instead? Then callers would only need to pass `ConfigContext.BROWSER`.
There'd still need to be two new keys for every new option, unless there is some `ConfigBoolWithContext` routine, though.

Reply:
Agreed that a single context enum would be better that passing multiple keys in. In the long term we may want something like ConfigBoolWithContext, but until we see 1-2 more similar uses, I'd be tempted to just have the backend fn look up specific keys based on the provided context.

Reply:
One downside of this new approach that I neglected to consider is that the frontend still needs to know the key names so it can look up the previously-set values. I'll add a separate method for fetching the defaults, so the calling code only has to think about the context, and not individual keys.

I'd also been thinking that the context would be specific to this method, instead of one that is shared with other routines. My reply didn't make that clear - apologies for the confusion. The problem with a shared context is that different routines may have different valid values. Say we have a different action in the future that wants a Context.REVIEWER and Context.PREVIEWER. If we have various routines sharing the same context, we will lose some of the advantages of enums, as then we run the risk of the calling code accidentally passing in an enum variant that a routine doesn't support, and we don't get notified about it at compile time.

If you feel things should be done differently, please let me know - I'm making these changes now to get a beta build out, and we can change them again if necessary.